src/main/java/duke/task/Deadline.java

```diff
@@ -3,24 +3,58 @@
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
 
+/**
+ * Deadline is a Task with a deadline.
+ * Contains the due date of task.
+ *
+ * @author Cheong Yee Ming
+ * @version Duke A-JavaDoc
+ */
 public class Deadline extends Task {
 
     protected final LocalDateTime by;
```

Comment:
A name like `by` is not really descriptive enough. Maybe consider something like `deadline`.